### PR TITLE
BUILD-3064 migrate to CirrusCI & Vault

### PR DESCRIPTION
**Pull request by julien-carsique-sonarsource**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    ARTIFACTORY_DEPLOY_USERNAME: vault-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer
+    ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer access_token]
+    SIGN_KEY: VAULT[development/kv/data/sign data.key]
+    PGP_PASSPHRASE: VAULT[development/kv/data/sign data.passphrase]
+    DEPLOY_PULL_REQUEST: true
+  <<: *SETUP_MAVEN_CACHE
+  build_script: |
+    source cirrus-env BUILD-PRIVATE
+    /etc/init.d/rng-tools-debian start
+    mvn org.eclipse.tycho:tycho-versions-plugin:1.5.0:set-version -Dtycho.mode=maven -DnewVersion=7.13.0.$BUILD_NUMBER -DgenerateBackupPoms=false -e
```
[the remark this answers is not in the document]
Comment (julien-carsique-sonarsource) [reply .cirrus.yml:82]: It's a draft, I was testing if we can reconcile tycho-versions-plugin and versions-maven-plugin called by set_maven_build_version...

[the remark this answers is not in the document]
Comment (thahnen) [reply .cirrus.yml:82]: Alrighty then 😄

**Comment by thahnen** [comment]
Would it be possible to remove the annoying Cirrus issues raised on the build script? The ones with conditions and such? Do they persist on all upcoming builds / PRs as annotations?

**Comment by julien-carsique-sonarsource** [comment]
> Would it be possible to remove the annoying Cirrus issues raised on the build script? The ones with conditions and such? Do they persist on all upcoming builds / PRs as annotations?

The Cirrus CI linter complain about only_if condition? 
It's a known defect that we took note of and we'll ask for a fix 
https://xtranet-sonarsource.atlassian.net/wiki/spaces/RE/pages/2501410819/Support+and+Feature+Requests#Noisy-false-positive-warnings-reported-by-the-linter

**Review by henryju — APPROVED** [review]

**Pull request merged**